src/services/SessionRequestProcessor.ts

```diff
@@ -135,6 +135,13 @@ export class SessionRequestProcessor {
   		return unauthorizedResponse;
   	}
 
+  	// Validate the address format and user details of the shared_claims received from the jwt.
+  	const { errorMsg, errorMsgCode } = this.validationHelper.isSharedClaimDataValid(jwtPayload);
```

Comment:
The checks that we've added to DocumentSelection look almost identical to those in the `isSharedClaimDataValid` function. What is the difference?

Reply:
What do you think about extending `isSharedClaimDataValid` so that it can be reused by DocumentSelection as well?

Reply:
Refactored.